ikasaneip/recovery-manager/src/main/java/org/ikasan/recovery/ScheduledRecoveryManager.java

```diff
@@ -134,6 +134,7 @@
     private Set<Object> recoveringIdentifiers = new HashSet<>();
 
     private boolean isConsumerMultiThreaded = false;
+    private boolean isEventBaseRecovery = false;
```

Comment:
This variable has the possibility of being set on the call to recovery with no event on one thread, then being flipped by a call to recovery with event on another thread (in a multi-threaded flow. Overeall I think it all works out, but worth being aware of.